### DISCUSSION TITLE
Best way to validate select list while also using an onselectionchange event

### DISCUSSION THREAD

**Discussion by GaryWayneSmith** [Megabit/Blazorise#5738] [General]
We have a need to reset a flag on field change for downstream validation.  We use the following code for text boxes to update our internal flag. This works fine and doesn't interfere with validation.  However, when we use the same onchange/oninput events on the select list, validation always fails. Switching the select box over to onselectionchanged does not actually seem to fire the event but validation works.  What's the best way to be able to validate with a select box while being able to handle an additional call to the onchange? 

```
@onchange="@((e)=>OnTextChanged(e))"
@oninput="@((e)=>OnTextChanged(e))"
```

```
<Validation Validator="@ValidationRule.IsSelected">
	<Select TValue="string"
		@bind-SelectedValue="@Model.StateOrProvince">
		<ChildContent>
			<SelectItem Value="@(string.Empty)">@(string.Empty)</SelectItem>
			@if (_listOfStates != null)
				@foreach (var state in _listOfStates)
				{
					<SelectItem Value="@state.Key">@state.Value</SelectItem>
				}
		</ChildContent>
		<Feedback>
			<ValidationNone>Please select value.</ValidationNone>
			<ValidationSuccess>Value is good.</ValidationSuccess>
			<ValidationError>Select valid value!</ValidationError>
		</Feedback>
	</Select>
</Validation>
```

This it the method being called. It's more of a test method right now with other code waiting to be implemented.
```
private async Task OnTextChanged(ChangeEventArgs args)
{
	Model.IsValidated = false;
	_addressOverride = false;
	await Task.CompletedTask;
}
```

**Comment by stsrki** [comment]
Validation fields usually all work separately, on their own. While you can hack it a little, it is not how it was designed so it will never be perfect. The only way I can think of that could help you to achieve to be "aware" of other fields is to use `IValidatableObject`

See the code in our demo for reference https://github.com/Megabit/Blazorise/blob/master/Demos/Blazorise.Demo/Pages/Tests/ValidationsPage.razor#L764

---

PS. By overriding

```
@onchange="@((e)=>OnTextChanged(e))"
@oninput="@((e)=>OnTextChanged(e))"
```

On a TextEdit you're basically skipping TextEdit internal logic and validation process.

**Comment by GaryWayneSmith** [comment]
I think the challenge that I have is that I only want to set my flag when the value has changed, but I always want to validate.  I'm not sure how the custom validator would help in that case. The custom validator link you send me is useful for other things.

Here is my use case:
We are collecting a full contact address.  Once we have received the address we validate it with the shipping carrier.  If it's good, we set a flag that it's been certified.  If the user changes any of the address data we need to reset the flag to not certified.

With that use case, what would be the best approach for the select list?  All of the other validations on the page are working fine.

**Reply by GaryWayneSmith** [reply]
I'm overthinking the problem.  I just need to switch to manual binding and handle it there.

```
SelectedValue="@Model.StateOrProvince"
SelectedValueChanged="@(async (e) => await OnStateOrProvinceChanged(e))"

// ...
	private async Task OnStateOrProvinceChanged(string newState)
	{
		Model.StateOrProvince = newState;
		Model.IsValidated = false;
		_addressOverride = false;
		await Task.CompletedTask;
	}

```

The the method OnStateOrProvinceChanged handles my logic for setting the flags and then sets the Model.StateOrProvice value. This allows the validation to work as expected.  I'm just used to using the @bind-* for handling the normal case.
